src/core/init/stall_avoider.ts

```diff
@@ -168,15 +182,46 @@ export default function StallAvoider(
 
   let prevFreezingState : { attemptTimestamp : number } | null;
 
-  return clock$.pipe(
+  /**
+   * If we're rebuffering waiting on data of a "locked stream", seek into the
+   * Period handled by that stream to unlock the situation.
+   */
+  const unlock$ = lockedStream$.pipe(
+    withLatestFrom(clock$),
+    tap(([lockedStreamEvt, tick]) => {
+      // TODO(PaulB) also skip when the user's wanted speed is set to `0`, as we
+      // might not want to seek in that case?
+      if (
+        !tick.rebuffering ||
+        tick.paused || (
+          lockedStreamEvt.bufferType !== "audio" &&
+          lockedStreamEvt.bufferType !== "video"
+        )
+      ) {
+        return;
+      }
+      const currPos = tick.position;
+      const rebufferingPos = tick.rebuffering.position ?? currPos;
+      const lckdPeriodStart = lockedStreamEvt.period.start;
```

Comment:
This is very minor but the name of the variable for saying the same thing is different "locked" vs "lckd".

Anyway, I think I have well understood the PR! I'm Happy

Reply:
Yes, I  kind of did it on purpose initially to reduce that variable name size while not sacrificing readability much, but considering this is only 1 letter this is a little ridiculous.

I update this now.